microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/Main.java

```diff
@@ -58,13 +60,21 @@ private Main() {
      * @param args command line arguments, currently ignored
      */
     public static void main(String[] args) {
-        CONTAINER.start();
+        if (ContainerInstanceHolder.isReset()) {
+            // in case somebody restarted the container, we need to get a new one
+            inUse = ContainerInstanceHolder.get();
+        } else {
+            // use the statically initialized one
+            inUse = CONTAINER;
+        }
+
+        inUse.start();
```

Comment:
I don't see the whole picture, but can someone call `reset()` on container holder between the `isReset()` check and `start()`?

Reply:
No. You cannot run two containers in parallel. You can run them sequentially. And this is to make sure that if you run a container, shut it down and then run the `Main` class, everything works fine (the `CONTAINER` field is there for `native-image` static initialization)